### DISCUSSION TITLE
[Question] Is infinity faster than sentence_transformers for inference?

### DISCUSSION THREAD

**Discussion by GingerNg** [michaelfeil/infinity#327] [Announcements]

**Comment by greptile-apps** [comment]
To compare the inference speed of Infinity and SentenceTransformers, follow these steps:

1. **Setup Benchmark Environment**: Use the provided `simple_app.py` script in `/docs/benchmarks/` to benchmark both Infinity and SentenceTransformers.

2. **Configure Environment Variables**:
   - Set `BENCHMARK_NAME` to `infinity` or `sentence_transformers`.
   - Set `DEVICE` to `cpu` or `cuda`.
   - Set `BATCH_SIZE` as needed.

3. **Run Benchmark**:
   - For Infinity: `BENCHMARK_NAME=infinity DEVICE=cpu BATCH_SIZE=32 python simple_app.py`
   - For SentenceTransformers: `BENCHMARK_NAME=sentence_transformers DEVICE=cpu BATCH_SIZE=32 python simple_app.py`

4. **Measure Inference Time**: Use the `/embeddings` endpoint to measure the time taken for embedding requests.

Refer to `/libs/infinity_emb/tests/end_to_end/test_optimum_embedding.py` for additional test cases and configurations.


### References
[/.github/ISSUE_TEMPLATE/new-model-addition.yml](https://github.com/michaelfeil/infinity/blob/main/.github/ISSUE_TEMPLATE/new-model-addition.yml)
[/.github/ISSUE_TEMPLATE/bug-report.yml](https://github.com/michaelfeil/infinity/blob/main/.github/ISSUE_TEMPLATE/bug-report.yml)
[/.github/ISSUE_TEMPLATE](https://github.com/michaelfeil/infinity/blob/main/.github/ISSUE_TEMPLATE)
[/libs/infinity_emb/tests/end_to_end/test_optimum_embedding.py](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/tests/end_to_end/test_optimum_embedding.py)
[/docs/docs/index.md](https://github.com/michaelfeil/infinity/blob/main/docs/docs/index.md)
[/docs/assets/openapi.json](https://github.com/michaelfeil/infinity/blob/main/docs/assets/openapi.json)
[/README.md](https://github.com/michaelfeil/infinity/blob/main/README.md)
[](https://github.com/michaelfeil/infinity/blob/main)
[/docs/docs](https://github.com/michaelfeil/infinity/blob/main/docs/docs)
[/docs/benchmarks/simple_app.py](https://github.com/michaelfeil/infinity/blob/main/docs/benchmarks/simple_app.py)
[/docs/docs/deploy.md](https://github.com/michaelfeil/infinity/blob/main/docs/docs/deploy.md)
[/docs](https://github.com/michaelfeil/infinity/blob/main/docs)
[/docs/benchmarks](https://github.com/michaelfeil/infinity/blob/main/docs/benchmarks)
[/docs/assets](https://github.com/michaelfeil/infinity/blob/main/docs/assets)
[/libs/infinity_emb](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb)
[/libs/infinity_emb/tests/data/benchmark](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/tests/data/benchmark)
[/libs/infinity_emb/tests/conftest.py](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/tests/conftest.py)
[/docs/docs/integrations.md](https://github.com/michaelfeil/infinity/blob/main/docs/docs/integrations.md)
[/libs/infinity_emb/infinity_emb/transformer/classifier](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/infinity_emb/transformer/classifier)
[/libs/infinity_emb/infinity_emb/fastapi_schemas/convert.py](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/infinity_emb/fastapi_schemas/convert.py)
[/libs/infinity_emb/tests/end_to_end/test_authentication.py](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/tests/end_to_end/test_authentication.py)
[/libs/infinity_emb/tests/unit_test](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/tests/unit_test)
[/libs/infinity_emb/tests](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/tests)
[/libs/infinity_emb/infinity_emb/fastapi_schemas](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/infinity_emb/fastapi_schemas)
[/libs/infinity_emb/Makefile](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/Makefile)

<details>
<summary>

#### About Greptile
</summary>

This response provides a starting point for your research, not a precise solution.

Help us improve! Please leave a 👍 if this is helpful and 👎 if it is irrelevant.

[Ask Greptile](https://app.greptile.com/chat/github/michaelfeil/infinity/main) · [Edit Issue Bot Settings](https://app.greptile.com/apps/github)
</details>

**Comment by michaelfeil** [comment]
Yes, https://michaelfeil.github.io/infinity/main/benchmarking/ For transformers 4.39.2. 

But it gets much faster if you implement no batching in your app.
